### PR TITLE
feat(RHINENG-2266): Add more task status states & informative messages

### PR DESCRIPTION
**Pull request by mhuth**
Screenshot showing the changes:
Screenshot from 2024-03-21 20-59-34

### CODE REVIEW

**Review by AsToNlele — APPROVED** [review]
Looks good! :smile: 
Is somewhere defined what the popover text should be? Couldn't find in the ticket
Make sure to update the snapshots so the CI can pass
`npm run test -- -u`

**Comment by AsToNlele** [comment]
LGTM, feel free to merge

**Comment by mhuth** [comment]
@AsToNlele thanks for the reviews.  I'm just waiting for the associated tasks backend changes to be finalized & merged first before I merge this one.

File: src/SmartComponents/ActivityTable/Columns.js
Comment (johnsonm325) [inline src/SmartComponents/ActivityTable/Columns.js:59]: I spoke with Andy Burka, and he needs to define these `message`s for each of the statuses. For the most part, they will be "canned"; but a task with a failure will need to use a message returned from the backend that gives more details of the failure.

**Review by johnsonm325 — CHANGES_REQUESTED** [review]
@mhuth Let's not merge this card until the messages get defined by Andy.

Most importantly, if a task fails to run (a failure), then the backend will need to return an error message. Specifically, something "human-readable" that explains in the popover, what failed.

File: src/SmartComponents/ActivityTable/Columns.js
Comment (PaulWay) [reply src/SmartComponents/ActivityTable/Columns.js:59]: Do we want to try to include the number of jobs that have failed / are still running / have completed?

File: src/SmartComponents/ActivityTable/Columns.js
Comment (mhuth) [reply src/SmartComponents/ActivityTable/Columns.js:83]: Fair enough, but sometimes I wonder if close enough is good enough.

File: src/SmartComponents/ActivityTable/Columns.js
Comment (mhuth) [reply src/SmartComponents/ActivityTable/Columns.js:59]: The message for a failed task may be different for each job that ran as part of the task.  Which one do we pick?  Also, we might not really know why a job failed other than the last thing we logged about it before it failed.  We don't get good error reporting back from Playbook Dispatcher.  Perhaps a canned message might be enough to let the user know they need to drill into each failed job to find out why each failed?

Comment (johnsonm325) [reply src/SmartComponents/ActivityTable/Columns.js:59]: I want to make sure we aren't confusing Failed with Completed with errors. If I understand correctly, Failed means that the task failed to run on any system. Meaning that it failed at the highest level. I would assume this means that the tasks backend failed to run the script, or the script failed before it ever communicated with a system. I'm not sure if this is even possible, so any feedback is appreciated.

Let me know what you think about this ^

**Comment by mhuth** [comment]
'Completed with errors' popover:
Screenshot from 2024-03-29 13-17-16

'Failure' popover:
Screenshot from 2024-03-29 13-17-37

Filtering on Failure and Completed With Errors:
Screenshot from 2024-03-29 13-20-07

File: src/SmartComponents/ActivityTable/Columns.js
Comment (johnsonm325) [inline src/SmartComponents/ActivityTable/Columns.js:74]: I think you can simply this like so:

```
const getStatusProps = (task) => {
  switch (task.status) {
    case TASK_STATUS.RUNNING:
      return {
        badgeColor: '#2B9AF3',
        iconStatus: 'info',
        IconType: InProgressIcon,
        popoverBodyContent: [
          `${task.running_jobs_count || 0} Running`,
          `${task.completed_jobs_count || 0} Completed`,
        ],
        hasPopover: true,
      };
    case TASK_STATUS.COMPLETED:
      return {
        badgeColor: '#3E8635',
        iconStatus: 'success',
        IconType: CheckCircleIcon,
        hasPopover: false,
      };
    case TASK_STATUS.COMPLETED_WITH_ERRORS:
      return {
        badgeColor: '#FF0000',
        iconStatus: 'danger',
        IconType: ExclamationCircleIcon,
        popoverBodyContent: [
          `${task.failure_jobs_count || 0} Failed`,
          `${task.timeout_jobs_count || 0} Timeout`,
          `${task.running_jobs_count || 0} Running`,
          `${task.completed_jobs_count || 0} Completed`,
        ],
        hasPopover: true,
      };
    case TASK_STATUS.FAILURE:
      return {
        badgeColor: '#FF0000',
        iconStatus: 'danger',
        IconType: ExclamationCircleIcon,
        popoverBodyContent:
          'All jobs in this task failed to run on their target systems.',
        hasPopover: true,
      };
    case TASK_STATUS.CANCELLED:
      return {
        badgeColor: '#FF0000',
        iconStatus: 'danger',
        IconType: ExclamationCircleIcon,
        popoverBodyContent: 'All jobs in this task have been cancelled',
        hasPopover: true,
      };
  }
};
```

Comment (johnsonm325) [reply src/SmartComponents/ActivityTable/Columns.js:74]: generally speaking, we want to avoid setting variables with `let` when possible. It's not always possible, but here I think is a good trade off.

File: src/SmartComponents/ActivityTable/Columns.js
Comment (johnsonm325) [inline src/SmartComponents/ActivityTable/Columns.js:126]: We should use this component to handle the underline: https://www.patternfly.org/components/description-list/#term-help-text

**Review by johnsonm325 — CHANGES_REQUESTED** [review]
I'm double checking with Andy about what the text color should be here. I don't think the mock-ups are right.
image

**Comment by mhuth** [comment]
> > I'm double checking with Andy about what the text color should be here. I don't think the mock-ups are right. image
> 
> The text should be darker than the icon in both the popover header and in the status column cell. The color should be danger color 100.

I'm not sure this is particularly important is it?

**Comment by mhuth** [comment]
Removed statuses with 0 counts from the popovers.  A 0 doesn't seem to provide any useful information to the customer:

Screenshot from 2024-04-02 11-22-56
Screenshot from 2024-04-02 11-22-51

**Comment by johnsonm325** [comment]
> > > I'm double checking with Andy about what the text color should be here. I don't think the mock-ups are right. image
> > 
> > 
> > The text should be darker than the icon in both the popover header and in the status column cell. The color should be danger color 100.
> 
> Is this particularly important?

Yes. This is a requirement of UX to ensure that we are being consistent across all apps.

**Comment by johnsonm325** [comment]
@mhuth Here's how the colors should be like what is shown here: https://www.patternfly.org/components/popover#alert-popover

This might make it easier to develop since there is already a component built for exactly this.

**Comment by johnsonm325** [comment]
Lastly, the "Running" popover needs to look like this from the mocks
image

**Comment by johnsonm325** [comment]
> > Lastly, the "Running" popover needs to look like this from the mocks image
> 
> Personally, I think mine looks better :)

Not necessarily disagreeing. I asked Andy which he wanted and UX gets the final say in design. 😀

**Comment by mhuth** [comment]
> @mhuth Here's how the colors should be like what is shown here: https://www.patternfly.org/components/popover#alert-popover
> 
> This might make it easier to develop since there is already a component built for exactly this.

Ok, I set the font color to #a30000, this is the correct font color for 'var(--pf-v5-c-popover--m-danger__title-text--Color)' ... I checked!

**Comment by mhuth** [comment]
@johnsonm325 Ok, I've made those latest changes, however, I refuse to make any more changes :)

**Review by johnsonm325 — APPROVED** [review]
This looks good. Thanks for all of the hard work on this!

**Comment by mhuth** [comment]
> This looks good. Thanks for all of the hard work on this!

And thank you @johnsonm325 for your help and patience.  Every frontend issue I work on is a learning experience for me.

**Comment by jiridostal** [comment]
@mhuth Thanks for the contributions, appreciate it! 
Re: Accurate colors, paddings etc - In some cases it may seem there is a better design implementation that those suggested by UXD, but we try to be accurate as there are some guidelines that get reviewed from time to time and these inconsistencies result in new tickets landing in our backlog, so we try to avoid it as much as possible :-) I would suggest discussing it directly with UX so have an approval on any discrepancies with mockups.

**Comment by mhuth** [comment]
> @mhuth Thanks for the contributions, appreciate it! Re: Accurate colors, paddings etc - In some cases it may seem there is a better design implementation that those suggested by UXD, but we try to be accurate as there are some guidelines that get reviewed from time to time and these inconsistencies result in new tickets landing in our backlog, so we try to avoid it as much as possible :-) I would suggest discussing it directly with UX so have an approval on any discrepancies with mockups.

@jiridostal, apologies as I was just getting a bit impatient to get the PR merged and the feature done, and was thinking 'close enough is good enough'.  TBH I didn't even notice the difference in colors and was thinking 'Would customers even notice, or care, too?'.  In future I'll certainly be mindful of getting it accurate as per the mocks so it doesn't result in subsequent 'nitpick jiras' being created that make more work for your team.  Again, apologies for my impatience and lack of understanding.

**Pull request merged**
